backpack/core/derivatives/embedding.py

```diff
@@ -0,0 +1,72 @@
+"""Derivatives for Embedding."""
+from typing import List, Tuple
+
+from torch import Tensor, einsum, zeros
+from torch.nn import Embedding
+
+from backpack.core.derivatives.basederivatives import BaseParameterDerivatives
+from backpack.utils import TORCH_VERSION_AT_LEAST_1_9_0
+from backpack.utils.subsampling import subsample
+
+
+class EmbeddingDerivatives(BaseParameterDerivatives):
+    """Derivatives for Embedding.
+
+    Note:
+        These derivatives assume the batch axis to be at position 0.
+
+    Index convention:
+    v - free axis
+    n - batch axis
+    s - num_embeddings
+    h - embedding_dim
+    """
+
+    def _jac_t_mat_prod(
+        self,
+        module: Embedding,
+        g_inp: Tuple[Tensor],
+        g_out: Tuple[Tensor],
+        mat: Tensor,
+        subsampling: List[int] = None,
+    ) -> Tensor:
+        raise NotImplementedError(
+            "Derivative w.r.t. input not defined: Input to Embedding has type long."
+            " But only float and complex dtypes can require gradients in PyTorch."
+        )
+
+    def _weight_jac_t_mat_prod(
+        self,
+        module: Embedding,
+        g_inp: Tuple[Tensor],
+        g_out: Tuple[Tensor],
+        mat: Tensor,
+        sum_batch: bool = True,
+        subsampling: List[int] = None,
+    ) -> Tensor:
+        self._check_parameters(module)
+
+        input0 = subsample(module.input0, subsampling=subsampling)
+        delta = zeros(module.num_embeddings, *input0.shape, device=mat.device)
+        for s in range(module.num_embeddings):
+            delta[s] = input0 == s
+        if TORCH_VERSION_AT_LEAST_1_9_0:
+            equation = f"sn...,vn...h->v{'' if sum_batch else 'n'}sh"
+        else:
+            equation = f"snx,vnxh->v{'' if sum_batch else 'n'}sh"
+            delta = delta.flatten(start_dim=2, end_dim=-1)
```

Comment:
tests fail here

Reply:
my bad. 
Fixed it by reintroducing the if-else branch. Now the condition is `if delta.dim() == 2`.